File: tests/compile-fail/ForeignNullableGenericConstructedBaseSlot.kt
// #86 — the crossing at the implementing position, on a CONSTRUCTED generic base.
//
// The slot is only recognisable in the frame the deriving type constructs it in: the reflected declaration says
// `Put(!0, List<int?>)` and the override says `Put(String, List<object>)`, so a comparison made against the OPEN
// declaration disagrees at the type variable and lets the uninhabitable override through. The supertype's arguments
// are substituted first, exactly as the override-slot bridge substitutes them.
import plainnet.GBase
import System.Collections.Generic.List

class CConstructed : GBase<String>() {
    override fun Put(tag: String, values: List<Int?>): String = "K:" + tag
}

fun main() {
    println(CConstructed().toString().substring(0, 2))
}

File: tests/compile-fail/ForeignNullableGenericConstructedBaseSlot.cs
using System;
using System.Collections.Generic;

// The crossing slot on a GENERIC .NET base, beside a parameter typed by the base's own type variable. Reflection
// hands back the OPEN declaration (`Put(!0, List<int?>)`); the Kotlin class derives from `GBase<String>` and states
// `Put(String, List<object>)`. Compared open, the two disagree at the type variable, the override is not recognised
// as filling the slot, and the class compiles clean and dies at load.
namespace plainnet
{
    public class GBase<T>
    {
        public virtual string Put(T tag, List<int?> values) => "net";
    }
}
